x/protorev/keeper/posthandler.go

```diff
@@ -0,0 +1,123 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+)
+
+type SwapToBackrun struct {
+	PoolId        uint64
+	TokenOutDenom string
+	TokenInDenom  string
+}
+
+type ProtoRevDecorator struct {
```

Comment:
Is ProtoRevDecorator implementing an Interface? If so, should we make that explicit? 

var _ Interface = Struct{} 

Even though it's unnecessary for the go compiler, it's a very common pattern in Tendermint. I don't know if it's widely used in Osmosis or in the SDK, but it's definitely helpful for helping your text editor find interface problems prior to compilation. No need to do this if it's uncommon.

Reply:
In this particular case, we are not implementing an interface. We did implement an interface in epoch_hooks.go which does include the one liner. For reference, the way the posthandler is implemented mirrors the same way the ante handler for other modules was designed (ex. https://github.com/osmosis-labs/osmosis/blob/2ec476bc12d2c8726c6850496ced274a5ba7ab49/x/txfees/keeper/feedecorator.go#L144).